src/BreakpointViewer.cpp

```diff
@@ -396,20 +401,19 @@ void BreakpointViewer::changeTableItem(BreakpointRef::Type type, QTableWidgetIte
 
 			if (auto range = parseLocationField(index, type, item->text(), combo ? combo->currentText() : "")) {
 				auto [begin, end] = *range;
-				QString address = QString("%1%2%3")
-				        .arg(hexValue(begin, adrLen))
-				        .arg(end ? ":" : "")
-				        .arg(end ? hexValue(*end, adrLen) : "");
-				setTextField(type, row, BP_ADDRESS, address);
 
 				// Use a symbolic address in the location field if available
-				QString location = ((type == BreakpointRef::BREAKPOINT) && debugSession.symbolTable().getAddressSymbol(item->text()))
-				                 ? item->text() : address;
-				setTextField(type, row, LOCATION, location, location != address ? address : "");
+				if (type == BreakpointRef::BREAKPOINT) {
+					Symbol* s = debugSession.symbolTable().getAddressSymbol(begin);
+					setTextField(type, row, LOCATION,
+					             s ? s->text() : locationString(begin, end, adrLen),
+					             s ? locationString(begin, end, adrLen) : "");
+				} else {
+					setTextField(type, row, LOCATION, locationString(begin, end, adrLen));
+				}
```

Comment:
When there are two or more labels with the same numerical value, and the user enters one of those label-names, this implementation might change that text to the name of a different label (with the same value).

I know this contradicts my earlier desire that the BreakpointViewer should be 'stateless' (all breakpoint state stored in openMSX itself (or in DebuggerForm)). But with this detail, BreakpointViewer 'remembers' which of the (equivalent) label-names the user entered. (Maybe it's acceptable that this info gets lost on a openMSX disconnect/reconnect, but changing the label immediately after the user entered it isn't very user friendly).

One thing I don't really like in this snippet is that you repeat the sub-expression `locationString(begin, end, adrLen)` 3 times in 4 lines. But it's not too long, so maybe it's acceptable.
(Maybe extract this sub-expression into a local variable before the `if`? Because you anyway will need it.)